crates/kitsune2_showcase/src/app/file_op_store.rs

```diff
@@ -0,0 +1,166 @@
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+};
+
+use bytes::Bytes;
+use kitsune2_api::{
+    BoxFut, Builder, Config, DhtArc, DynOpStore, DynOpStoreFactory, K2Error,
+    K2Result, MetaOp, OpId, OpStore, OpStoreFactory, SpaceId, Timestamp,
+};
+use kitsune2_core::factories::{MemOpStoreFactory, MemoryOpRecord};
+
+use super::file_data::FileData;
+
+pub type FileStoreLookup = Arc<Mutex<HashMap<String, OpId>>>;
+
+#[derive(Debug)]
+pub struct FileOpStoreFactory {
+    mem_op_store: MemOpStoreFactory,
+    file_store_lookup: FileStoreLookup,
+}
+
+impl FileOpStoreFactory {
+    pub fn create(file_store_lookup: FileStoreLookup) -> DynOpStoreFactory {
+        let out: DynOpStoreFactory = Arc::new(FileOpStoreFactory {
+            mem_op_store: MemOpStoreFactory {},
+            file_store_lookup,
+        });
+        out
+    }
+}
+
+impl OpStoreFactory for FileOpStoreFactory {
+    fn default_config(&self, config: &mut Config) -> K2Result<()> {
+        self.mem_op_store.default_config(config)
+    }
+
+    fn validate_config(&self, config: &Config) -> K2Result<()> {
+        self.mem_op_store.validate_config(config)
+    }
+
+    fn create(
+        &self,
+        builder: Arc<Builder>,
+        space: SpaceId,
+    ) -> BoxFut<K2Result<DynOpStore>> {
+        Box::pin(async move {
+            let file_store_lookup = self.file_store_lookup.clone();
+            let out: DynOpStore = Arc::new(FileOpStore {
+                mem_op_store: self.mem_op_store.create(builder, space).await?,
+                file_store_lookup,
+            });
+            Ok(out)
+        })
+    }
+}
```

Comment:
Wrap the MemOpStore in an Arc to make it clone-able, then:

```suggestion
    fn create(
        &self,
        builder: Arc<Builder>,
        space: SpaceId,
    ) -> BoxFut<'static, K2Result<DynOpStore>> {
        let file_store_lookup = self.file_store_lookup.clone();
        let mem_op_store = self.mem_op_store.clone();
        Box::pin(async move {
            let out: DynOpStore = Arc::new(FileOpStore {
                mem_op_store: mem_op_store.create(builder, space).await?,
                file_store_lookup,
            });
            Ok(out)
        })
    }
}
```

Reply:
ammended